### PR TITLE
TST: Add test groupby transform no column

### LINKED ISSUE #37093: BUG: groupby-transform produces NaN for series keys with as_index=False
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
df = pd.DataFrame({"X": [1.0]})
df.groupby(pd.Series(["group"]), as_index=False).transform("sum")
```

#### Problem description

With version 1.1.3 this produces a dataframe containing NaN.

#### Expected Output

Expected is 1.0 and this is also what older versions produce (1.0.5).

It seems the following conditions have to be satisfied to trigger this bug:

- grouping by a series which does not contain indices corresponding to the original dataframe (grouping by `pd.Series([0])` works though it seems accidental)
- using `as_index=False`
- using `transform`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.6.6.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.76-linuxkit
Version          : #1 SMP Tue May 26 11:42:35 UTC 2020
machine          : x86_64
processor        :
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.2.3
setuptools       : 41.2.0
Cython           : 0.29.15
pytest           : 5.4.3
hypothesis       : 5.6.0
sphinx           : 1.8.5
blosc            : None
feather          : None
xlsxwriter       : 1.2.8
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : 2.8.5 (dt dec pq3 ext lo64)
jinja2           : 2.10.3
IPython          : 7.9.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.6.2
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.17.1.post2
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.2.1
sqlalchemy       : 1.3.15
tables           : None
tabulate         : 0.8.6
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by DevpriyaDave**
- [X] closes #37093 
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/groupby/transform/test_transform.py [hunk 1]
```diff
@@ -650,6 +650,16 @@ def test_groupby_transform_with_nan_group():
     expected = Series([1.0, 1.0, 2.0, 3.0, np.nan, 6.0, 6.0, 9.0, 9.0, 9.0], name="a")
     tm.assert_series_equal(result, expected)
 
+def test_groupby_transform_with_no_column():
+    # GH 49864
+    df = pd.DataFrame({0: [1.0]})
+    gb = df.groupby(pd.Series(["a"]), as_index=False)
+    msg = "A grouping was used that is not in the columns of the DataFrame"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        result = gb.sum()
```
Comment (rhshadrach) [inline pandas/tests/groupby/transform/test_transform.py:659]: For the linked issue, you want to be using `.transform("sum")` here.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by DevpriyaDave** [comment]
> Thanks for the PR!

Thanks for your review and appreciate you updating the PR title, I will make a note of the format for next time.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**